### ISSUE TITLE
Setup auto-updates

### ISSUE THREAD

**Issue by poorgeek** [ThePacielloGroup/CCAe#46]
Electron apps have their own built-in method of auto-updating using Squirrel. Would be nice to set this up so that updates are automatic rather than manual.

**Comment by poorgeek** [comment]
Found a [guide for setting up auto-updates](https://medium.com/heresy-dev/auto-updating-apps-for-windows-and-osx-using-electron-the-complete-guide-4aa7a50b904c) that may be useful.

**Comment by ferllings** [comment]
For ref, there is a bug in the last version of electron-updater: https://github.com/electron-userland/electron-builder/issues/3269

Sticking to v3.0.4 for now

**Comment by poorgeek** [comment]
The auto updater appears to be working correctly on Mac. Was able to update from the last beta to 1.0.0 final today.

One thing to note is that the text of the Update dialog could be improved.

<img width="239" alt="CCA update dialog" src="https://user-images.githubusercontent.com/489299/47235583-ba5cd000-d396-11e8-97be-5f1c1a746a0b.png">

The dialog reads as:

> v1.0.0 downloaded, application will be quit for update...

Can we change that to something like:

> v1.0.0 has been downloaded. Select Ok to restart with the new version.

Or something like that. @patrickhlauke ?

**Comment by patrickhlauke** [comment]
will do a PR in a bit :)

**Comment by patrickhlauke** [comment]
on windows, i get this rather intimidating error dialog...

![ccae-updater-win-error](https://user-images.githubusercontent.com/895831/47241472-b60ad400-d3e3-11e8-96db-eac2e0ec9c2f.png)

tl;dr: some issue with the signed executable/installer

**Comment by patrickhlauke** [comment]
unfortunately, i can't quite remember how i installed the version i currently have on my machine. i'd imagine that i had downloaded it from github though...

**Comment by patrickhlauke** [comment]
additionally, it might be worth instead of a dialog to show some discreet little extra line that gives users the option to update now or just carry on with their work and having it install next time they fire up the application (otherwise it'll end up being like Filezilla...every time i fire it up - every few months - to get something done it immediately pesters me about installing the latest update, which is disruptive when you started it just to get something done quickly).

see for instance how github app does it...

![ccae-updater-win-error](https://user-images.githubusercontent.com/895831/47241615-3df0de00-d3e4-11e8-9cee-809e89bd1695.png)

**Comment by patrickhlauke** [comment]
just tested to confirm: uninstalled whichever version of CCAe i had, grabbed the previous release beta from github. the above error appears when it first starts and finds the update.

**Comment by poorgeek** [comment]
@patrickhlauke with all of your testing of the recent 1.1.0 release, are you still seeing this issue?

**Comment by patrickhlauke** [comment]
not seen this error reappear, no. however, i did not test the actual autoupdating, just uninstalled my current version and then installed the latest version...will have a look if i can force an update from an older one.

**Comment by patrickhlauke** [comment]
installed the 1.0.0 setup.exe. it finds the update, and then the error happens as before...

> ![Capture](https://user-images.githubusercontent.com/895831/55824621-1f021780-5afc-11e9-99c3-772ecfa2d379.PNG)

**Comment by patrickhlauke** [comment]
with the CCA-1.1.0 setup, it doesn't show the error message right away. when i explicitly go to "Install update" menu option, however, a different error appears

> ![Capture](https://user-images.githubusercontent.com/895831/55824813-8cae4380-5afc-11e9-9829-539e43b0faba.PNG)

**Comment by ferllings** [comment]
I'm closing this as it seems to be gone. Feel free to re-open it if it comes back
